docs/how-to/type-mapping.rst

```diff
@@ -76,13 +76,16 @@ keys). :class:`ObjCStrInstance` also handles Unicode code points above
 :class:`NSString` is based on UTF-16.
 
 If you have an :class:`ObjCStrInstance` instance, and you need to pass that
-instance to a method that does a specific typecheck for `str`, you can use
-:class:`str(nsstring)` to convert the :class:`ObjCStrInstance` instance to
+instance to a method that does a specific typecheck for :class:`str`, you can
+use ``str(nsstring)`` to convert the :class:`ObjCStrInstance` instance to
 :class:`str`.
 
 :class:`ObjCStrInstance` implements all the utility methods that are available
-on :class:`str`, such as ``replace`` and ``split``. These utility methods all
-return *Python* strings.
+on :class:`str`, such as ``replace`` and ``split``. When these methods return
+a string, the implementation may return Python :class:`str` or
+:class:`ObjCStrInstance` instances. If you need to use the return value from
+these methods, you should always use ``str()`` to ensure you have a Python
+string.
```

Comment:
Maybe we should explain the opposite case as well: if you want to call a `NSString` method on the returned string, you need to wrap it in `at` first. Admittedly this case is less common; more often you're passing the string back into an Objective-C method as an argument, and in that case `str` is converted to `NSString` automatically.

Reply:
Done. I've also added some example code to clarify the text.